File: src/main/java/com/github/redfoos/logstash/psi/LogstashTokenType.java
package com.github.redfoos.logstash.psi;

import com.github.redfoos.logstash.LogstashLanguage;
import com.intellij.psi.tree.IElementType;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;

public class LogstashTokenType extends IElementType {
    public LogstashTokenType(@NotNull @NonNls String debugName) {
        super(debugName, LogstashLanguage.getINSTANCE());
    }

    @Override
    public String toString() {
        return "Logstash token type." + super.toString();
    }
}

File: src/main/kotlin/com/github/redfoos/logstash/psi/LogstashTokenType.kt
package com.github.redfoos.logstash.psi

import com.github.redfoos.logstash.LogstashLanguage
import com.intellij.psi.tree.IElementType

class LogstashTokenType(debugName: String) : IElementType(debugName, LogstashLanguage.INSTANCE) {

    override fun toString(): String {
        return "Logstash token type.${super.toString()}"
    }
}
